### PR TITLE
backend/googlephotos: Add start_year for google photos options

### PR DESCRIPTION
**Pull request by dooven**
#### What is the purpose of this change?
Make the start year configurable instead of hard coding it to `2000`

#### Was the change discussed in an issue or in the forum before?
Discussion: https://github.com/rclone/rclone/issues/3630

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by dooven** [comment]
Im getting an error while runnign the integration test. I might have reached my quota for uploading photos

```
--- FAIL: TestIntegration (1.73s)
    --- FAIL: TestIntegration/CreateAlbum (1.64s)
        --- FAIL: TestIntegration/CreateAlbum/PutFile (0.12s)
            googlephotos_test.go:60: 
                        Error Trace:    googlephotos_test.go:60
                        Error:          Received unexpected error:
                                        {
                                          "code": 8,
                                          "message": "Quota exceeded for quota metric \u0027Size of uploads\u00
27 and limit \u0027Size of uploads per day\u0027 of service \u0027photoslibrary.googleapis.com\u0027 for consum
er \u0027project_number:202264815644\u0027."
                                        } (429 429 Too Many Requests)
                                        couldn't upload file
                                        github.com/rclone/rclone/backend/googlephotos.(*Object).Update
                                                /home/daven/go/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:978
                                        github.com/rclone/rclone/backend/googlephotos.(*Fs).Put
                                                /home/daven/go/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:636
                                        github.com/rclone/rclone/backend/googlephotos.TestIntegration.func1.1
                                                /home/daven/go/src/github.com/rclone/rclone/backend/googlephotos/googlephotos_test.go:59
                                        testing.tRunner
                                                /usr/local/go/src/testing/testing.go:909
                                        runtime.goexit
                                                /usr/local/go/src/runtime/asm_amd64.s:1357
                        Test:           TestIntegration/CreateAlbum/PutFile
    --- PASS: TestIntegration/UploadMkdir (0.00s)
        --- PASS: TestIntegration/UploadMkdir/List (0.00s)
        --- PASS: TestIntegration/UploadMkdir/Rmdir (0.00s)
        --- PASS: TestIntegration/UploadMkdir/ListEmpty (0.00s)
    --- FAIL: TestIntegration/Upload (0.09s)
        googlephotos_test.go:225: 
                Error Trace:    googlephotos_test.go:225
                Error:          Received unexpected error:
                                {
                                  "code": 8,
                                  "message": "Quota exceeded for quota metric \u0027Size of uploads\u0027 and limit \u0027Size of uploads per day\u0027 of service \u0027photoslibrary.googleapis.com\u0027 for consumer \u0027project_number:202264815644\u0027."
                                } (429 429 Too Many Requests)
                                couldn't upload file
                                github.com/rclone/rclone/backend/googlephotos.(*Object).Update
                                        /home/daven/go/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:978
                                github.com/rclone/rclone/backend/googlephotos.(*Fs).Put
                                        /home/daven/go/src/github.com/rclone/rclone/backend/googlephotos/googlephotos.go:636
                                github.com/rclone/rclone/backend/googlephotos.TestIntegration.func3
                                        /home/daven/go/src/github.com/rclone/rclone/backend/googlephotos/googlephotos_test.go:224
                                testing.tRunner
                                        /usr/local/go/src/testing/testing.go:909
                                runtime.goexit
                                        /usr/local/go/src/runtime/asm_amd64.s:1357
```

File: backend/googlephotos/googlephotos.go [hunk 1]
```diff
@@ -134,14 +134,22 @@ rclone mount needs to know the size of files in advance of reading
 them, so setting this flag when using rclone mount is recommended if
 you want to read the media.`,
 			Advanced: true,
-		}},
+		},
+			{
```
Comment (ncw) [inline backend/googlephotos/googlephotos.go:138]: Can you use the same bracketty style as the other options please? So `}, {` and `}},`

**Comment by dooven** [comment]
> That is looking good - see small nit inline.
> 
> As for integration tests - you will probably have to get your own client_id as rclone's seems to be overloaded somehow :-( Or wait 24hours anyway!

Hi. I created my own client_id and looks like the integration passes :+1: 

```
=== RUN   TestNewAlbums
--- PASS: TestNewAlbums (0.00s)
=== RUN   TestAlbumsAdd
--- PASS: TestAlbumsAdd (0.00s)
=== RUN   TestAlbumsDel
--- PASS: TestAlbumsDel (0.00s)
=== RUN   TestAlbumsGet
--- PASS: TestAlbumsGet (0.00s)
=== RUN   TestAlbumsGetDirs
--- PASS: TestAlbumsGetDirs (0.00s)
=== RUN   TestIntegration
=== RUN   TestIntegration/CreateAlbum
=== RUN   TestIntegration/CreateAlbum/PutFile
=== RUN   TestIntegration/CreateAlbum/PutFile/ObjectFs
=== RUN   TestIntegration/CreateAlbum/PutFile/ObjectString
=== RUN   TestIntegration/CreateAlbum/PutFile/ObjectHash
=== RUN   TestIntegration/CreateAlbum/PutFile/ObjectSize
=== RUN   TestIntegration/CreateAlbum/PutFile/ObjectSetModTime
=== RUN   TestIntegration/CreateAlbum/PutFile/ObjectStorable
=== RUN   TestIntegration/CreateAlbum/PutFile/ObjectOpen
=== RUN   TestIntegration/CreateAlbum/PutFile/CheckFileInAlbum
=== RUN   TestIntegration/CreateAlbum/PutFile/CheckInByYear
=== RUN   TestIntegration/CreateAlbum/PutFile/CheckInByMonth
=== RUN   TestIntegration/CreateAlbum/PutFile/CheckInByDay
=== RUN   TestIntegration/CreateAlbum/PutFile/NewObject
=== RUN   TestIntegration/CreateAlbum/PutFile/NewObjectWithID
=== RUN   TestIntegration/CreateAlbum/PutFile/NewFsIsFile
=== RUN   TestIntegration/CreateAlbum/PutFile/RemoveFileFromAlbum
=== RUN   TestIntegration/UploadMkdir
=== RUN   TestIntegration/UploadMkdir/List
=== RUN   TestIntegration/UploadMkdir/Rmdir
=== RUN   TestIntegration/UploadMkdir/ListEmpty
=== RUN   TestIntegration/Upload
=== RUN   TestIntegration/Upload/List
=== RUN   TestIntegration/Upload/NewObject
=== RUN   TestIntegration/Upload/NewObjectWithID
=== RUN   TestIntegration/Name
=== RUN   TestIntegration/Root
=== RUN   TestIntegration/String
=== RUN   TestIntegration/Features
=== RUN   TestIntegration/Precision
=== RUN   TestIntegration/Hashes
--- PASS: TestIntegration (17.91s)
    --- PASS: TestIntegration/CreateAlbum (14.55s)
        --- PASS: TestIntegration/CreateAlbum/PutFile (12.80s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/ObjectFs (0.00s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/ObjectString (0.00s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/ObjectHash (0.00s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/ObjectSize (0.92s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/ObjectSetModTime (0.00s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/ObjectStorable (0.00s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/ObjectOpen (0.08s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/CheckFileInAlbum (0.53s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/CheckInByYear (0.56s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/CheckInByMonth (0.43s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/CheckInByDay (0.66s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/NewObject (0.54s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/NewObjectWithID (0.33s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/NewFsIsFile (1.52s)
            --- PASS: TestIntegration/CreateAlbum/PutFile/RemoveFileFromAlbum (3.00s)
    --- PASS: TestIntegration/UploadMkdir (0.00s)
        --- PASS: TestIntegration/UploadMkdir/List (0.00s)
        --- PASS: TestIntegration/UploadMkdir/Rmdir (0.00s)
        --- PASS: TestIntegration/UploadMkdir/ListEmpty (0.00s)
    --- PASS: TestIntegration/Upload (3.36s)
        --- PASS: TestIntegration/Upload/List (0.00s)
        --- PASS: TestIntegration/Upload/NewObject (0.00s)
        --- PASS: TestIntegration/Upload/NewObjectWithID (0.40s)
    --- PASS: TestIntegration/Name (0.00s)
    --- PASS: TestIntegration/Root (0.00s)
    --- PASS: TestIntegration/String (0.00s)
    --- PASS: TestIntegration/Features (0.00s)
    --- PASS: TestIntegration/Precision (0.00s)
    --- PASS: TestIntegration/Hashes (0.00s)
=== RUN   TestAddID
--- PASS: TestAddID (0.00s)
=== RUN   TestFileAddID
--- PASS: TestFileAddID (0.00s)
=== RUN   TestFindID
--- PASS: TestFindID (0.00s)
=== RUN   TestPatternMatch
=== RUN   TestPatternMatch/#0,root="",itemPath="",isFile=false
=== RUN   TestPatternMatch/#1,root="",itemPath="",isFile=true
=== RUN   TestPatternMatch/#2,root="upload",itemPath="",isFile=false
=== RUN   TestPatternMatch/#3,root="upload/dir",itemPath="",isFile=false
=== RUN   TestPatternMatch/#4,root="upload/file.jpg",itemPath="",isFile=true
=== RUN   TestPatternMatch/#5,root="media",itemPath="",isFile=false
=== RUN   TestPatternMatch/#6,root="",itemPath="media",isFile=false
=== RUN   TestPatternMatch/#7,root="media/all",itemPath="",isFile=false
=== RUN   TestPatternMatch/#8,root="media",itemPath="all",isFile=false
=== RUN   TestPatternMatch/#9,root="media/all",itemPath="file.jpg",isFile=true
--- PASS: TestPatternMatch (0.00s)
    --- PASS: TestPatternMatch/#0,root="",itemPath="",isFile=false (0.00s)
    --- PASS: TestPatternMatch/#1,root="",itemPath="",isFile=true (0.00s)
    --- PASS: TestPatternMatch/#2,root="upload",itemPath="",isFile=false (0.00s)
    --- PASS: TestPatternMatch/#3,root="upload/dir",itemPath="",isFile=false (0.00s)
    --- PASS: TestPatternMatch/#4,root="upload/file.jpg",itemPath="",isFile=true (0.00s)
    --- PASS: TestPatternMatch/#5,root="media",itemPath="",isFile=false (0.00s)
    --- PASS: TestPatternMatch/#6,root="",itemPath="media",isFile=false (0.00s)
    --- PASS: TestPatternMatch/#7,root="media/all",itemPath="",isFile=false (0.00s)
    --- PASS: TestPatternMatch/#8,root="media",itemPath="all",isFile=false (0.00s)
    --- PASS: TestPatternMatch/#9,root="media/all",itemPath="file.jpg",isFile=true (0.00s)
=== RUN   TestPatternMatchToEntries
=== RUN   TestPatternMatchToEntries/#0,root="",itemPath=""
=== RUN   TestPatternMatchToEntries/#1,root="upload",itemPath=""
=== RUN   TestPatternMatchToEntries/#2,root="upload",itemPath="dir"
=== RUN   TestPatternMatchToEntries/#3,root="media",itemPath=""
=== RUN   TestPatternMatchToEntries/#4,root="media/all",itemPath=""
=== RUN   TestPatternMatchToEntries/#5,root="media",itemPath="all"
=== RUN   TestPatternMatchToEntries/#6,root="media/by-year",itemPath=""
=== RUN   TestPatternMatchToEntries/#7,root="media/by-year/2000",itemPath=""
=== RUN   TestPatternMatchToEntries/#8,root="media/by-month",itemPath=""
=== RUN   TestPatternMatchToEntries/#9,root="media/by-month/2001",itemPath=""
=== RUN   TestPatternMatchToEntries/#10,root="media/by-month/2001/2001-01",itemPath=""
=== RUN   TestPatternMatchToEntries/#11,root="media/by-day",itemPath=""
=== RUN   TestPatternMatchToEntries/#12,root="media/by-day/2001",itemPath=""
=== RUN   TestPatternMatchToEntries/#13,root="media/by-day/2001/2001-01-02",itemPath=""
=== RUN   TestPatternMatchToEntries/#14,root="album",itemPath=""
=== RUN   TestPatternMatchToEntries/#15,root="album/sub",itemPath=""
=== RUN   TestPatternMatchToEntries/#16,root="album/sub/one",itemPath=""
=== RUN   TestPatternMatchToEntries/#17,root="shared-album",itemPath=""
=== RUN   TestPatternMatchToEntries/#18,root="shared-album/sub",itemPath=""
=== RUN   TestPatternMatchToEntries/#19,root="shared-album/sub/one",itemPath=""
--- PASS: TestPatternMatchToEntries (0.00s)
    --- PASS: TestPatternMatchToEntries/#0,root="",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#1,root="upload",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#2,root="upload",itemPath="dir" (0.00s)
    --- PASS: TestPatternMatchToEntries/#3,root="media",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#4,root="media/all",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#5,root="media",itemPath="all" (0.00s)
    --- PASS: TestPatternMatchToEntries/#6,root="media/by-year",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#7,root="media/by-year/2000",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#8,root="media/by-month",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#9,root="media/by-month/2001",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#10,root="media/by-month/2001/2001-01",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#11,root="media/by-day",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#12,root="media/by-day/2001",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#13,root="media/by-day/2001/2001-01-02",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#14,root="album",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#15,root="album/sub",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#16,root="album/sub/one",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#17,root="shared-album",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#18,root="shared-album/sub",itemPath="" (0.00s)
    --- PASS: TestPatternMatchToEntries/#19,root="shared-album/sub/one",itemPath="" (0.00s)
=== RUN   TestPatternYears
--- PASS: TestPatternYears (0.00s)
=== RUN   TestPatternMonths
--- PASS: TestPatternMonths (0.00s)
=== RUN   TestPatternDays
--- PASS: TestPatternDays (0.00s)
=== RUN   TestPatternYearMonthDayFilter
--- PASS: TestPatternYearMonthDayFilter (0.00s)
=== RUN   TestPatternAlbumsToEntries
--- PASS: TestPatternAlbumsToEntries (0.00s)
PASS
ok      github.com/rclone/rclone/backend/googlephotos   17.927
```

**Comment by ncw** [comment]
Super job - well done :-)

I'll merge this now - thank you!

**Comment by dooven** [comment]
Np. Thanks for this awesome repo @ncw !

**Pull request merged**